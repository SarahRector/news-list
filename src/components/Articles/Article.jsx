import React from 'react';
import PropTypes from 'prop-types';

const Article = ({ title, author, description }) => (
  <>
    <h1>{title}</h1>
    <p>By: {author}</p>
    <p>Summary: {description}</p>
  </>
);

Article.propTypes = {
  title: PropTypes.string.isRequired,
  author: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired
};

export default Article;
