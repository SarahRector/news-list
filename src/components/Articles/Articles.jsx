import React from 'react';
import PropTypes from 'prop-types';
import Article from './Article';

const Articles = ({ articles }) => {
  const articleElements = articles.map(article => (
    <li key={article.title}>
      <Article {...article} />
    </li>
  ));

  return (
    <ul data-testid="articles">
      {articleElements}
    </ul>
  );
};

Articles.propTypes = {
  articles: PropTypes.arrayOf(PropTypes.shape({
    title: PropTypes.string.isRequired,
    author: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired
  })).isRequired
};

export default Articles;
