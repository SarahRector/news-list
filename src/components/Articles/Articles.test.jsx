/* eslint-disable max-len */
import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Articles from './Articles';

const testArray = [
  { title: 'Wow!', author: 'Bob Barker', description: 'These are words about things.' },
  { title: 'You\'re Wearing That?', author: 'Anne Hathaway', description: 'I guess if that\'s the impression you want to give.' },
  { title: 'Why Are You Doing It THAT Way?', author: 'Brandon Roy', description: 'I\'m just saying, I usually do it this other way and it\'s better.' },
];

describe('Articles component', () => {
  afterEach(() => cleanup());
  it('renders Articles', () => {
    const { asFragment } = render(<Articles
      articles={testArray}
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
