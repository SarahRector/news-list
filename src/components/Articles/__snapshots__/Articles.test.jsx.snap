// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Articles component renders Articles 1`] = `
<DocumentFragment>
  <ul
    data-testid="articles"
  >
    <li>
      <h1>
        Wow!
      </h1>
      <p>
        By: Bob Barker
      </p>
      <p>
        Summary: These are words about things.
      </p>
    </li>
    <li>
      <h1>
        You're Wearing That?
      </h1>
      <p>
        By: Anne Hathaway
      </p>
      <p>
        Summary: I guess if that's the impression you want to give.
      </p>
    </li>
    <li>
      <h1>
        Why Are You Doing It THAT Way?
      </h1>
      <p>
        By: Brandon Roy
      </p>
      <p>
        Summary: I'm just saying, I usually do it this other way and it's better.
      </p>
    </li>
  </ul>
</DocumentFragment>
`;
